packages/astro/components/Code.astro

```diff
@@ -43,7 +50,10 @@ function repairShikiTheme(html: string): string {
 }
 
 const highlighter = await shiki.getHighlighter({ theme });
-const _html = highlighter.codeToHtml(code, { lang });
+if(typeof lang !== 'string') await highlighter.loadLanguage(lang)
```

Comment:
is it okay if this runs on every codeblock render? If not, can you do a check that the language hasn't already been loaded?

Reply:
The language is loaded to the highlighter instance. Because the instance is being generated on every render, so does the `loadLanguage` function. I'll see if I can optimize it a bit more